tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs

```diff
@@ -0,0 +1,146 @@
+                AssertLoggedTelemetry(result, eventDetails, TokenSource.IdentityProvider, CacheRefreshReason.NotApplicable);
+
+                // Acquire token silently
+                var account = (await _pca.GetAccountsAsync().ConfigureAwait(false)).Single();
+                result = await _pca.AcquireTokenSilent(TestConstants.s_scope, account).ExecuteAsync().ConfigureAwait(false);
+                Assert.IsNotNull(result);
+
+                eventDetails = ((TelemetryClient)_telemetryClient).TestTelemetryEventDetails;
+                AssertLoggedTelemetry(result, eventDetails, TokenSource.Cache, CacheRefreshReason.NotApplicable);
+            }
+        }
+
+        private void AssertLoggedTelemetry(AuthenticationResult authenticationResult, MsalTelemetryEventDetails eventDetails, TokenSource tokenSource, CacheRefreshReason cacheRefreshReason)
+        {
+            Assert.IsNotNull(eventDetails);
+            Assert.AreEqual(Convert.ToInt64(cacheRefreshReason), eventDetails.Properties[TelemetryConstants.CacheInfoTelemetry]);
+            Assert.AreEqual(Convert.ToInt64(tokenSource), eventDetails.Properties[TelemetryConstants.TokenSource]);
+            Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationTotalInMs, eventDetails.Properties[TelemetryConstants.Duration]);
+            Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs, eventDetails.Properties[TelemetryConstants.DurationInHttp]);
+            Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs, eventDetails.Properties[TelemetryConstants.DurationInCache]);
+            Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationTotalInMs, eventDetails.Properties[TelemetryConstants.Duration]);
+        }
+
+        private PublicClientApplication CreateApplication()
+        {
+            PublicClientApplication pca = PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                    .WithHttpManager(_harness.HttpManager)
+                    .WithDefaultRedirectUri()
+                    .WithExperimentalFeatures()
+                    .WithTelemetryClient(_telemetryClient)
+                    .BuildConcrete();
+
+            TokenCacheHelper.PopulateCache(_pca.UserTokenCacheInternal.Accessor);
+
+            return pca;
+        }
+    }
+
+    internal class TelemetryClient : ITelemetryClient
+    {
+        public MsalTelemetryEventDetails TestTelemetryEventDetails { get; set; }
+        
+        public TelemetryClient(string clientId)
+        {
+            ClientId = clientId;
+        }
+
+        public string ClientId { get; set; }
+
+        public void Initialize()
+        {
+
+        }
+
+        public bool IsEnabled()
+        {
+            return true;
+        }
+
+        public bool IsEnabled(string eventName)
+        {
+            return TelemetryConstants.AcquireTokenEventName.Equals(eventName) ? true : false;
```

Comment:
>? true : false; [](http://example.com/codeflow?start=78&length=15)

no need to add the extra ternary statement, Equals already returns what you want right?